accounts/abi/abi.go

```diff
@@ -135,11 +136,12 @@ func (abi *ABI) UnmarshalJSON(data []byte) error {
 				_, ok = abi.Methods[name]
 			}
 			abi.Methods[name] = Method{
-				Name:    name,
-				RawName: field.Name,
-				Const:   field.Constant,
-				Inputs:  field.Inputs,
-				Outputs: field.Outputs,
+				Name:            name,
+				RawName:         field.Name,
+				Const:           field.Constant,
+				StateMutability: field.StateMutability,
```

Comment:
It looks like geth sets this on the `"constructor"` case too: https://github.com/ethereum/go-ethereum/blob/master/accounts/abi/abi.go#L143